### PR TITLE
Implement memory allocation for aarch64 Linux

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

File: Headers/DebugServer2/Utils/HexValues.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ template <typename T> static inline std::string ToHex(T const &vec) {
   std::string result;
   for (char n : vec) {
     result += NibbleToHex(n >> 4);
-    result += NibbleToHex(n & 15);
+    result += NibbleToHex(n & 0x0f);
```
Comment (a20012251) [inline Headers/DebugServer2/Utils/HexValues.h]: It'd be nice to define a uint4_t with static casts :3

File: Sources/Host/Linux/ARM64/PTraceARM64.cpp [hunk 2]
```diff
@@ -60,20 +60,11 @@ int PTrace::getMaxHardwareWatchpoints(ProcessThreadId const &ptid) {
 ErrorCode PTrace::readCPUState(ProcessThreadId const &ptid,
                                ProcessInfo const &pinfo,
                                Architecture::CPUState &state) {
-  pid_t pid;
-
-  ErrorCode error = ptidToPid(ptid, pid);
```
Comment (a20012251) [inline Sources/Host/Linux/ARM64/PTraceARM64.cpp]: probably useful at some point

Comment (xiaobai) [inline Sources/Host/Linux/ARM64/PTraceARM64.cpp]: I figured. It's weird that it was left in if it wasn't being used, though.

Comment (sas) [inline Sources/Host/Linux/ARM64/PTraceARM64.cpp]: Yeah I think that's when I was doing the initial support for aarch64, I used the ARM version as a template and never realized that the pid wasn't used in the ARM64 version.

File: Headers/DebugServer2/Utils/HexValues.h (hunk 1 above)
Comment (sas) [inline Headers/DebugServer2/Utils/HexValues.h]: I think the goal here is to just be a little bit more explicit about what we're doing. Having a uint4_t would be cute but I'm not sure I want to implement that right now :p

File: Headers/DebugServer2/Architecture/ARM64/CPUState.h [hunk 3]
```diff
@@ -38,10 +38,10 @@ typedef ds2::Architecture::ARM::CPUState CPUState32;
 //
 struct CPUState64 {
   union {
-    uint64_t regs[32 + 1 + 1];
+    uint64_t regs[31 + 1 + 1 + 1];
```
Comment (sas) [inline Headers/DebugServer2/Architecture/ARM64/CPUState.h]: That's just a semi-useful way of keeping track of where these registers come from. There are 31 GP registers (x0...x30) and sp, pc, and cpsr. This change doesn't have any actual impact on how the code runs.

Comment (xiaobai) [inline Headers/DebugServer2/Architecture/ARM64/CPUState.h]: 👍

**Pull request merged**